include/boost/fusion/adapted/boost_pfr/tag_of_fallback.hpp

```diff
@@ -0,0 +1,94 @@
+#include <boost/fusion/support/tag_of_fwd.hpp>
+#include <boost/type_index/ctti_type_index.hpp>
+#include <boost/pfr/traits.hpp>
+#include <type_traits>
+
+namespace boost { namespace pfr {
+  struct boost_fusion_tag;
+}} // namespace boost::pfr
+
+namespace boost { namespace fusion
+{
+    struct boost_pfr_tag;
+    struct non_fusion_tag;
+    struct fusion_sequence_tag;
+
+    namespace detail
+    {
+        // Helper function that returns true if `name` starts with `substr`
+        template <std::size_t N>
+        constexpr bool starts_with(const char* name, const char (&substr)[N]) noexcept {
+            static_assert(N > 0, "non a zero-terminated string passed");
+            for (std::size_t i = 0; i<N-1; ++i) {
+                if (name[i] != substr[i]) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        // Function that returns true if `T` declared in namespace `ns`
+        // got from here: https://www.boost.org/doc/libs/1_81_0/doc/html/boost_typeindex/examples.html#boost_typeindex.examples.c_14_checking_namespace_at_compi
+        // TODO: test for it
+        template <class T, std::size_t N>
+        constexpr bool in_namespace(const char (&ns)[N]) noexcept {
+            const char* name = boost::typeindex::ctti_type_index::type_id<T>().raw_name();
+
+            // Some compilers add `class ` or `struct ` before the namespace, so we need to skip those words first
+            if (detail::starts_with(name, "class ")) {
+                name += sizeof("class ") - 1;
+            } else if (detail::starts_with(name, "struct ")) {
+                name += sizeof("struct ") - 1;
+            }
+
+            return detail::starts_with(name, ns) && detail::starts_with(name + N - 1, "::");
+        }
+
+        template<typename T>
+        struct constexpr_in_namespace_check_possible {
+            template <bool = detail::in_namespace<T>("") >
+            static std::true_type test(long) noexcept;
+
+            static std::false_type test(...) noexcept;
+
+            static constexpr bool value = decltype(test(0)){};
+        };
+
+        template<typename T>
+        constexpr bool is_implicitly_reflectable_via_pfr() noexcept {
+            // TODO: check that T is complete
+            const auto possible_pfr = boost::pfr::is_implicitly_reflectable<
+                T, boost::pfr::boost_fusion_tag>::value;
+            static_assert(detail::constexpr_in_namespace_check_possible<T>::value
+                        , "Unfortunately, Boost PFR with implicit reflection won't work in your compiler due to constexpr limitations. "
+                          "Please, try to use another compiler or disable implicit reflection by '-DBOOST_FUSION_PFR_ENABLE_IMPLICIT_REFLECTION=0'");
+            const auto value = !std::is_array<T>::value
+                          // FIXME do we need this?
+                          // && !std::is_reference<T>::value
```

Comment:
We needn't this. PFR doesn't work with references: https://godbolt.org/z/PEsWhf84c

Reply:
> PFR doesn't work with references

Maybe we should add a small "boost_pfr_reference" test to show it

Reply:
Added "boost_pfr_reference" test with a little bit limitations due to PFR side issue.